File: tests/fixtures/python/classes.py
from dataclasses import dataclass


class Base:
    """Base class."""

    LIMIT: int = 10

    def save(self) -> None:
        self.validate()

    def validate(self) -> None:
        pass


@dataclass
class User(Base):
    name: str = "anonymous"

    @property
    def display(self) -> str:
        return self.name.title()

File: tests/fixtures/typescript/classes.ts
@injectable()
class Repository extends BaseRepo implements Store {
  limit = 100;
  handler = (e: Event) => this.process(e);

  constructor(private db: Database) {
    super();
  }

  async save(user: User): Promise<void> {
    await this.db.write(user);
  }

  private process(e: Event): void {
    validate(e);
  }
}
